src/Lib.P2P.Tests/Routing/Dht1Test.cs

```diff
@@ -0,0 +1,478 @@
+                await dht.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task RemovesPeerFromRoutingTable()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var dht = new DhtService {SwarmService = swarm};
+            await dht.StartAsync();
+            try
+            {
+                var peer = swarm.RegisterPeerAddress(
+                    "/ip4/127.0.0.1/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h");
+                Assert.IsTrue(dht.RoutingTable.Contains(peer));
+
+                swarm.DeregisterPeer(peer);
+                Assert.IsFalse(dht.RoutingTable.Contains(peer));
+            }
+            finally
+            {
+                await dht.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task ProcessFindNodeMessage_Self()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var dht = new DhtService {SwarmService = swarm};
+            await dht.StartAsync();
+            try
+            {
+                var request = new DhtMessage
+                {
+                    Type = MessageType.FindNode,
+                    Key = self.Id.ToArray()
+                };
+                var response = dht.ProcessFindNode(request, new DhtMessage());
+                Assert.AreEqual(1, response.CloserPeers.Length);
+                var ok = response.CloserPeers[0].TryToPeer(out var found);
+                Assert.IsTrue(ok);
+                Assert.AreEqual(self, found);
+            }
+            finally
+            {
+                await dht.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task ProcessFindNodeMessage_InRoutingTable()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var dht = new DhtService {SwarmService = swarm};
+            await dht.StartAsync();
+            try
+            {
+                dht.RoutingTable.Add(other);
+                var request = new DhtMessage
+                {
+                    Type = MessageType.FindNode,
+                    Key = other.Id.ToArray()
+                };
+                var response = dht.ProcessFindNode(request, new DhtMessage());
+                Assert.AreEqual(1, response.CloserPeers.Length);
+                var ok = response.CloserPeers[0].TryToPeer(out var found);
+                Assert.IsTrue(ok);
+                Assert.AreEqual(other, found);
+                CollectionAssert.AreEqual(other.Addresses.ToArray(),
+                    found.Addresses.Select(a => a.WithoutPeerId()).ToArray());
+            }
+            finally
+            {
+                await dht.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task ProcessFindNodeMessage_InSwarm()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var other = swarm.RegisterPeerAddress(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename 'other' which hides the field with the same name.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)